velox/type/Variant.cpp

```diff
@@ -192,10 +192,32 @@ std::string variant::toJson(const TypePtr& type) const {
   // and other basic operations. Stringification logic should not be specific
   // to variants; it should be consistent for all map representations
 
+  static uint32_t totalIdxForVariantChild;
+  static uint32_t totalIdxForInputChild;
+
+  if ((kind_ == TypeKind::ROW) || (kind_ == TypeKind::ARRAY)) {
+    totalIdxForVariantChild = 0;
+    totalIdxForInputChild = 0;
+  }
+
   if (isNull()) {
     return "null";
   }
 
+  VELOX_USER_CHECK_EQ(
+      this->kind(),
+      type->kind(),
+      "variant TypeKind {} does not match input TypeKind {}",
+      this->kind(),
+      type->kind());
+
+  VELOX_USER_CHECK_EQ(
```

Comment:
these checks must go into ROW and Array toJson() function.

Reply:
Changed it to be used with ROW and ARRAY cases in toJson() function.